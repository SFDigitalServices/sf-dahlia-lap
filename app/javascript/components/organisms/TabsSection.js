import React from 'react'
import TabsMenu from '../molecules/TabsMenu'
import TabCard from './TabCard'

const TabsSection = ({ children, background, padding, ...props }) => {
  return (
    <>
      <div className={`tabs-section tabs-row full-width inner--3x bg-${background}`}>
        <TabsMenu {...props} />
      </div>
      <TabCard padding={padding}>
        {children}
      </TabCard>
    </>
  )
}

TabsSection.defaultProps = {
  background: 'snow'
}

export default TabsSection
