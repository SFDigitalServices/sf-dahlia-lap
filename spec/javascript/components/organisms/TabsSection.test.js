/* global mount */
import React from 'react'

import TabsSection from 'components/organisms/TabsSection'

describe('TabsMenu', () => {
  test('it should render correctly', () => {
    const items = [
      { title: 'Short Form Application', url: '/url1' },
      { title: 'Supplemental Information', url: '/url2', active: true }
    ]

    const wrapper = mount(<TabsSection items={items} currentUrl={'/url1'} />)

    expect(wrapper).toMatchSnapshot()
  })
})
