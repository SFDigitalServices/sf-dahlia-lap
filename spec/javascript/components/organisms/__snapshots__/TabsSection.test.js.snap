// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsMenu it should render correctly 1`] = `
<TabsSection
  background="snow"
  currentUrl="/url1"
  items={
    Array [
      Object {
        "title": "Short Form Application",
        "url": "/url1",
      },
      Object {
        "active": true,
        "title": "Supplemental Information",
        "url": "/url2",
      },
    ]
  }
>
  <div
    className="tabs-section tabs-row row full-width inner--3x bg-snow"
  >
    <TabsMenu
      currentUrl="/url1"
      items={
        Array [
          Object {
            "title": "Short Form Application",
            "url": "/url1",
          },
          Object {
            "active": true,
            "title": "Supplemental Information",
            "url": "/url2",
          },
        ]
      }
    >
      <ul
        className="tabs full-width-small-only"
        role="menubar"
      >
        <Tab
          key="Short Form Application"
          linkRefs={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          title="Short Form Application"
          url="/url1"
        >
          <li
            className="tab-title"
            role="none"
          >
            <a
              href="/url1"
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="0"
            >
              Short Form Application
            </a>
          </li>
        </Tab>
        <Tab
          active={true}
          key="Supplemental Information"
          linkRefs={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          title="Supplemental Information"
          url="/url2"
        >
          <li
            className="tab-title active"
            role="none"
          >
            <a
              aria-selected={true}
              href="/url2"
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1"
            >
              Supplemental Information
            </a>
          </li>
        </Tab>
      </ul>
    </TabsMenu>
  </div>
  <TabCard
    padding={false}
  >
    <section
      className="tabs-card-row row full-width inner--3x margin-bottom--2x"
    />
  </TabCard>
</TabsSection>
`;
